jscomp/core/lam_compile_global.ml

```diff
@@ -63,8 +64,9 @@ let expand_global_module  id env  : J.expression =
     (Lam_module_ident.of_ml id) 
     (Has_env env)
     ~not_found:(fun _ -> assert false)
-    ~found:(fun   {signature; _} -> 
-        Js_of_lam_module.make ~comment:id.name 
+    ~found:(fun   {signature; _} ->
+        let names = Ocaml_types.map (fun name -> name) signature in
+        Js_of_lam_module.make ~comment:id.name names
           (Ocaml_types.map (fun name -> E.ml_var_dot id name) signature )
       )
```

Comment:
If I understand correctly, after your changes, this expansion is not necessary any more, should be a nop

Reply:
Tried with `E.var id`. In principle it looks OK, though with that simple case, `id` is not required anymore in the generated code. Not sure why.

Reply:
That could be a bug that should be fixed, actually this is quite a large simplification after merging your patch, I will see how to fix it